sympy/functions/special/holonomic.py

```diff
@@ -0,0 +1,524 @@
+"""Holonomic Functions and Differential Operators"""
+
+from __future__ import print_function, division
+
+from sympy import symbols, Symbol, diff, S, Mul
+from sympy.polys.polytools import lcm, gcd
+from sympy.core.expr import Expr
+from sympy.printing import sstr
+from sympy.matrices import Matrix
+from sympy.core.compatibility import range
+from sympy.polys.polytools import DMP
+from sympy.core.function import Function
+
+
+def DiffOperatorAlgebra(base, generator):
+    """ A function to create a Differential Operator Algebra.
+    The first arguments need to be the base polynomial ring for the algebra
+    and the second argument must be a generator.
+    Returns the class representing the operator algebra and
+    the standard operator for differentiation i.e. the `Dx` operator.
+
+    Examples
+    =======
+    >>> from sympy.polys.domains import ZZ
+    >>> from sympy import symbols
+    >>> from sympy.functions.special.holonomic import DiffOperatorAlgebra
+    >>> x = symbols('x')
+    >>> R, Dx = DiffOperatorAlgebra(ZZ.old_poly_ring(x), 'Dx')
+    """
+
+    ring = DifferentialOperatorAlgebra(base, generator)
+    return (ring, ring.derivative_operator)
+
+
+class DifferentialOperatorAlgebra:
```

Comment:
Every class should subclass from `object`. Otherwise in Python 2 it will be an old-style class.